cuda_core/tests/test_program.py

```diff
@@ -9,10 +9,187 @@
 from cuda.core.experimental import _linker
 from cuda.core.experimental._module import Kernel, ObjectCode
 from cuda.core.experimental._program import Program, ProgramOptions
+from cuda.core.experimental._utils.cuda_utils import driver, handle_return
 
+cuda_driver_version = handle_return(driver.cuDriverGetVersion())
 is_culink_backend = _linker._decide_nvjitlink_or_driver()
 
 
+def _is_nvvm_available():
+    """Check if NVVM is available."""
+    try:
+        from cuda.core.experimental._program import _get_nvvm_module
+
+        _get_nvvm_module()
+        return True
+    except RuntimeError:
+        return False
+
+
+nvvm_available = pytest.mark.skipif(
+    not _is_nvvm_available(), reason="NVVM not available (libNVVM not found or cuda-bindings < 12.9.0)"
+)
+
+try:
+    from cuda.core.experimental._utils.cuda_utils import driver, handle_return
+
+    _cuda_driver_version = handle_return(driver.cuDriverGetVersion())
+except Exception:
+    _cuda_driver_version = 0
+
+_libnvvm_version = None
+_libnvvm_version_attempted = False
+
+precheck_nvvm_ir = """
+target triple = "nvptx64-nvidia-cuda"
+target datalayout = "e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-i128:128:128-f32:32:32-f64:64:64-v16:16:16-v32:32:32-v64:64:64-v128:128:128-n16:32:64"
+
+define void @dummy_kernel() {
+entry:
+  ret void
+}
+
+!nvvm.annotations = !{!0}
+!0 = !{void ()* @dummy_kernel, !"kernel", i32 1}
```

Comment:
Took me a while to debug...

This IR is incomplete (it fails at `verify_program()`) because it missed the `!nvvmir.version` attribute.

Reply:
```
IR version 1.0 incompatible with current version 2.0\nprecheck.ll: error: incompatible IR detected. Possible mix of compiler/IR from different releases.\x00
```